### PR TITLE
fix: Change applicationset generate HTTP method to avoid route conflicts

### LINKED ISSUE #20757: New applicationset generate endpoint conflicts with existing one
Checklist:

* [X] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [X] I've included steps to reproduce the bug.
* [X] I've pasted the output of `argocd version`.

**Describe the bug**

Introduced in https://github.com/argoproj/argo-cd/commit/3a5b653fb153edff37ca1460ef34d44301bf23a5, the `Generate` defines `post: /api/v1/applicationsets` which already exists as the `Create` rpc, breaking the existing API.

**To Reproduce**
```
curl -XPOST http://localhost:4000/api/v1/applicationsets --d '{"apiVersion":"argoproj.io/v1alpha1","kind":"ApplicationSet","metadata":{"name":"guestbook"},"spec":{"goTemplate":true,"goTemplateOptions":["missingkey=error"],"generators":[],"template":{"metadata":{"name":"guestbook"},"spec":{"project":"my-project","source":{"repoURL":"https://github.com/infra-team/cluster-deployments.git","targetRevision":"HEAD","path":"guestbook/foo"},"destination":{"server":"localhost","namespace":"guestbook"}}}}}'
Results in a 500 
```
{"error":"error creating ApplicationSets: ApplicationSets is nil in request","code":2,"message":"error creating ApplicationSets: ApplicationSets is nil in request"}%

**Expected behavior**
Should return 200 on a valid applicationset or 400 with an invalid 

**Screenshots**

**Version**

```shell
dist/argocd version  
argocd: v2.14.0+b39f3b1.dirty
  BuildDate: 2024-11-11T16:13:12Z
  GitCommit: b39f3b15fbf1abf59177b58b6e87a44e32600bbe
  GitTreeState: dirty
  GoVersion: go1.22.0
  Compiler: gc
  Platform: darwin/arm64

```

**Logs**

```
Paste any relevant application logs here.
```

### PR DESCRIPTION
**Pull request by amit-o**
Closes #20757
Fixes HTTP route conflict introduced in https://github.com/argoproj/argo-cd/commit/3a5b653fb153edff37ca1460ef34d44301bf23a5 
I'm not sure if `PUT` is the correct semantic we want here, however because `/api/v1/applicationsets/{name}` is already taken I wasn't sure to which route to change this to. Open to change to whatever.

If possible, would appreciate this as a hotfix to 2.13
Checklist:

* [X] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [X] The title of the PR states what changed and the related issues number (used for the release note).
* [X] The title of the PR conforms to the [Toolchain Guide](https://argo-cd.readthedocs.io/en/latest/developer-guide/toolchain-guide/#title-of-the-pr)
* [X] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I've updated documentation as required by this PR.
* [X] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md#legal)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).
* [ ] My new feature complies with the [feature status](https://github.com/argoproj/argoproj/blob/master/community/feature-status.md) guidelines.
* [X] I have added a brief description of why this PR is necessary and/or what this PR solves.
* [ ] Optional. My organization is added to USERS.md.
* [x] Optional. For bug fixes, I've indicated what older releases this fix should be cherry-picked into (this may or may not happen depending on risk/complexity).

### CODE REVIEW

**Review by andrii-korotkov-verkada — APPROVED** [review]

**Comment by amit-o** [comment]
Thanks @andrii-korotkov-verkada, is there someone we can ping for further approval?
This is a blocker for us, so I'd appreciate it if it can be hotfixed to 2.13

**Comment by andrii-korotkov-verkada** [comment]
I'll bring up the PRs marked with ready-for-review on the contributors meeting, hope they can take a look soon.

**Comment by amit-o** [comment]
Hey @andrii-korotkov-verkada, sorry for bothering, but as it's been a few contributors meetings since I'd like to ping this again

**Comment by andrii-korotkov-verkada** [comment]
I've put it as a my top item for the next meeting.

File: server/applicationset/applicationset.proto [hunk 1]
```diff
@@ -74,7 +74,7 @@ service ApplicationSetService {
 	// Generate generates
 	rpc Generate (ApplicationSetGenerateRequest) returns (ApplicationSetGenerateResponse) {
 		option (google.api.http) = {
-			post: "/api/v1/applicationsets"
+			put: "/api/v1/applicationsets"
```
Comment (agaudreault) [inline server/applicationset/applicationset.proto:77]: I think the convention is to use put for create, for other request that arent (crud), they should have a suffix.
```suggestion
			post: "/api/v1/applicationsets/generate"
```

**Review by agaudreault — CHANGES_REQUESTED** [review]

File: server/applicationset/applicationset.proto (hunk 1 above)
Comment (amit-o) [reply server/applicationset/applicationset.proto:77]: @agaudreault unless I'm mistaken, that will conflict with `GET /api/v1/applicationsets/{name}` assuming an applicationset is named `generate`. Maybe that's a tradeoff we're okay with taking, just making sure that's accepted

**Review by agaudreault — APPROVED** [review]
LGTM

**Pull request merged**